stumpy/core.py

```diff
@@ -589,6 +594,41 @@ def check_window_size(m, max_size=None):
     if max_size is not None and m > max_size:
         raise ValueError(f"The window size must be less than or equal to {max_size}")
 
+    if n is not None:
+        # Raise warning if there is at least one subsequence with no
+        # non-trivial neighbour in the case of a self-join.
+
+        # For any time series `T`, an "eligible nearest neighbor" subsequence for
+        # the central-most subsequence must be located outside the `excl_zone`,
+        # and the central-most subsequence will ALWAYS have the smallest gap
+        # to its furthest neighbour. Therefore, we only need to check whether
+        # the `excl_zone` eliminates all "neighbors" for the central-most subsequence
+        # in `T`. In fact, we just need to verify whether the `excl_zone` eliminates
+        # the "neighbor" that is furthest away (index-wise) from the central-most
+        # subsequence. If it does not, this implies that all subsequences in `T`
+        # will have at least one or more "eligible nearest neighbors" outside
+        # their respective `excl_zone`
+
+        excl_zone = int(math.ceil(m / config.STUMPY_EXCL_ZONE_DENOM))
+
+        l = n - m + 1
+
+        # If `l` is odd (`l == 2k+1`):
+        # The central subsequence is at index `k`, with furthest neighbors at `0`
+        # and `2k`, both `k == l // 2` indices away from the central-most subsequence.
+
+        # If `l` is even (`l == 2k`):
+        # The central subsequences are at `k-1` and `k`. The furthest neighbor is
+        # at `2k-1` for `k-1`, and `0` for `k`. In both cases, the subsequence
+        # and its furthest neighbor are `k == l // 2` indices away.
+        max_gap = l // 2
+        if max_gap <= excl_zone:
```

Comment:
`if farthest_idx <= excl_zone`

I think that communicates the idea very well now!

Reply:
Hmmm, I guess, technically, it's not really the "farthest index"? Saying "distance" doesn't feel right either because it gets confused with Euclidean distance.

Reply:
Ahh, I see that you've come to a similar conclusion in https://github.com/TDAmeritrade/stumpy/pull/1078#discussion_r2023928170 with "index-wise distance". Hmmm...

Reply:
Or `farthest_idx_diff`?

Reply:
I think I like `diff_to_farthest_idx`. It's long but, to your point, we've described it in the comment and we'll only use this variable once so it's not a big deal.